src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -0,0 +1,272 @@
+                                    assembledBreakpoint.homology = getSequence(contigsCollection, contigId, assembledBreakpoint.region2.startInContig - 1, assembledBreakpoint.region1.endInContig);
+                                }
+
+                                assembledBreakpoint.insertedSequence = "NA";
+                                if (assembledBreakpoint.region1.endInContig < assembledBreakpoint.region2.startInContig - 1) {
+                                    assembledBreakpoint.insertedSequence = getSequence(contigsCollection, contigId, assembledBreakpoint.region1.endInContig, assembledBreakpoint.region2.startInContig);
+                                }
+
+                                assembledBreakpoints.add(assembledBreakpoint);
+                            }
+                        }
+                    }
+                }
+        } catch (final IOException e) {
+            throw new GATKException("could not execute BWA");
+        }
+
+        return assembledBreakpoints;
+    }
+
+    /**
+     * Pulls the sequence of the requested contig between the start and end coordinates from the contigsCollection
+     * @param contigsCollection
+     * @param contigId
+     * @param startInContig
+     * @param endInContig
+     * @return
+     */
+    private String getSequence(final RunSGAViaProcessBuilderOnSpark.ContigsCollection contigsCollection, final String contigId, final int startInContig, final int endInContig) {
+        for (final Tuple2<RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigID, RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigSequence> contig : contigsCollection.getContents()) {
+            if (contig._1.equals(new RunSGAViaProcessBuilderOnSpark.ContigsCollection.ContigID(contigId))) {
+                final String sequence = contig._2.toString();
+                return sequence.substring(startInContig, endInContig);
+            }
+        }
+        throw new GATKException("Could not find contig " + contigId);
+    }
+
+    /**
+     * Wrap a contig sequence in a ShortRead object and pass it to BWA to align
+     * @param bwaMem
+     * @param contigId
+     * @param sequence
+     * @return
+     * @throws IOException
+     */
+    private AlnRgn[] bwaAlignSequence(final BwaMem bwaMem, final String contigId, final byte[] sequence) throws IOException {
+        final ShortRead contigShortRead = new ShortRead(contigId, sequence, qualSequence(sequence.length));
+        return bwaMem.align(contigShortRead);
+    }
+
+    /**
+     * Generate a bogus base quality sequence to pass in for the aligned contig (since the jBWA API requires that reads have qualities)
+     * @param length
+     * @return
+     */
+    private byte[] qualSequence(final int length) {
+        final byte[] quals = new byte[length];
+        for (int i = 0; i < quals.length; i++) {
+            quals[i] = 'A';
+        }
+        return quals;
+    }
+
+    public void close() {
+        bwaMem.dispose();
```

Comment:
It seems that the constructor constructed the `BwaIndex` object, so I believe `index.close();` is necessary here.